gcloud/datastore/key.py

```diff
@@ -83,9 +83,8 @@ def _validate_dataset_id(self):
                  can be implied.
         """
         if self._dataset_id is None:
-            if _implicit_environ.DATASET is not None:
-                # This assumes DATASET.id() is not None.
-                self._dataset_id = _implicit_environ.DATASET.id()
+            if _implicit_environ.DATASET_ID is not None:
```

Comment:
I have a sense that `_combine_args` and `_validate_dataset_id` should be combined.